github/resource_github_team_sync_group_mapping_test.go

```diff
@@ -0,0 +1,180 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccGithubTeamSyncGroupMapping_basic(t *testing.T) {
+	if isEnterprise != "true" {
+		t.Skip("Skipping because `ENTERPRISE_ACCOUNT` is not set or set to false")
+	}
+	randString := acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum)
+	teamName := fmt.Sprintf("tf-acc-test-%s", randString)
+	groupName := "test_team_group"
+	description := fmt.Sprintf("tf-group-description-%s", randString)
+	rn := "github_team_sync_group_mapping.test_mapping"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGithubTeamSyncGroupMappingDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGithubTeamSyncGroupMappingConfig(teamName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(rn, "group.#"),
+					resource.TestCheckResourceAttr(rn, "group.0.name", groupName),
+				),
+			},
+			{
+				Config: testAccGithubTeamSyncGroupMappingAddGroupAndUpdateConfig(teamName, description),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(rn, "group.#"),
+					resource.TestCheckResourceAttr(rn, "group.0.description", description),
+					resource.TestCheckResourceAttr(rn, "group.1.description", description),
```

Comment:
The test, as written, I think is only adding one group to the team.

Reply:
ah so for this check i was imagining the list comprehension `[for g in data.github_organization_team_sync_groups.test_groups.groups : g if "test_team_group" in g.group_name]`  could match more than one group if in the list of possible groups they existed as ["team_team_group_a", "test_team_group_b", ...]

Reply:
but i think it makes more sense to just include all groups like you mentioned in a comment above --which in that case perhaps these attribute checks would need to be in a loop to check the description on all the groups available?